pkg/api/addons.go

```diff
@@ -321,6 +324,16 @@ func (cs *ContainerService) setAddonsConfig(isUpdate bool) {
 		},
 	}
 
+	defaultKubeDNSAddonsConfig := KubernetesAddon{
+		Name:    KubeDNSAddonName,
+		Enabled: to.BoolPtr(!common.IsKubernetesVersionGe(o.OrchestratorVersion, "1.12.0")),
```

Comment:
this is only a default, I don't see the Enabled value getting overriden anywhere for > 1.12 upgrades and/or < 1.12 clusters that have a kube-dns addon defined in the apimodel with Enabled set to `false`.